apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -509,48 +520,51 @@ var WorkshopForm = React.createClass({
           <br/>
           <Row>
             <Col sm={4}>
-              <Input
-                type="text"
-                label="Location Name"
-                value={this.state.location_name || ''}
-                onChange={(event) => {this.handleFieldChange('location_name', event.target.value);}}
-                bsStyle={validation.style.location_name}
-                help={validation.help.location_name}
-                maxLength={255}
-                style={this.props.readOnly && styles.readOnlyInput}
-                disabled={this.props.readOnly}
-              />
+              <FormGroup validationState={validation.style.location_name}>
+                <ControlLabel>Location Name</ControlLabel>
+                <FormControl
+                  type="text"
+                  value={this.state.location_name || ''}
+                  onChange={(event) => {this.handleFieldChange('location_name', event.target.value);}}
+                  maxLength={255}
+                  style={this.props.readOnly && styles.readOnlyInput}
+                  disabled={this.props.readOnly}
+                />
+                <HelpBlock>{validation.help.location_name}</HelpBlock>
+              </FormGroup>
             </Col>
             <Col sm={6}>
-              <Input
-                type="text"
-                key={this.state.useAutocomplete} // Change key to force re-draw
-                className="location-autocomplete"
-                label="Location Address"
-                value={this.state.location_address || ''}
-                placeholder="Enter a location"
-                onChange={(event) => {this.handleFieldChange('location_address', event.target.value);}}
-                bsStyle={validation.style.location_address}
-                help={validation.help.location_address}
-                maxLength={255}
-                style={this.props.readOnly && styles.readOnlyInput}
-                disabled={this.props.readOnly}
-              />
+              <FormGroup validationState={validation.style.location_address}>
+                <ControlLabel>Location Address</ControlLabel>
+                <FormControl
+                  type="text"
+                  key={this.state.useAutocomplete} // Change key to force re-draw
+                  ref={(ref) => this.locationAddressControl = ReactDOM.findDOMNode(ref)}
+                  value={this.state.location_address || ''}
+                  placeholder="Enter a location"
+                  onChange={(event) => {this.handleFieldChange('location_address', event.target.value);}}
```

Comment:
I see. That makes sense. I was initially going with DRY over having a method for each field initially (#1 above), but I didn't think about the downside of re-creating all these functions in render.

I'll try the #2 approach to satisfy both requirements. I think I can set the `name` attribute on these inputs, and then inspect that in `onChange` via `$(e.target).attr('name')`